File: Engine/Models/Transaction.py
from marshmallow import Schema, fields
from Configuration.config import db

class Transaction(db.Model):
    __tablename__ = 'transaction'
    id = db.Column(db.Integer, primary_key=True, autoincrement=True) #
    sender = db.Column(db.String(64)) #account number that sent money
    receiver = db.Column(db.String(64)) #account number that received money
    amount = db.Column(db.Float) #amount of money sent in currency bellow
    currency = db.Column(db.String(3)) #currency
    state = db.Column(db.String(32)) #state of transaction (In progress, Approved, Denied)


    def __init__(self, sender, receiver, amount, currency, state):
        self.sender = sender
        self.receiver = receiver
        self.amount = amount
        self.currency = currency
        self.state = state

class TransactionSchema(Schema):
    id = fields.Number()
    sender = fields.Str()
    receiver = fields.Str()
    amount = fields.Float()
    currency = fields.Str()
    state = fields.Str()

File: UI/src/components/Transactions/Transaction.js
import React from "react";
import styles from "./Transaction.module.css";

const Transaction = (props) => {
  return (
    <li className={styles.transaction}>
      <span className={styles.sender}>{props.sender}</span>
      <span className={styles.amount}>{props.amount}</span>
      <span className={styles.currency}>{props.currency}</span>
      <span className={styles.state}>{props.state}</span>
    </li>
  );
};

export default Transaction;
